XLS-0081d-permissioned-dex/README.md

```diff
@@ -0,0 +1,593 @@
+
+### 5.1. Fields
+
+<details>
+<summary>
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/transactions/types/payment) are the existing fields for the `Payment` transaction.
+</summary>
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Amount`|✔️|`string`|`Amount`|The maximum amount of currency to deliver. For non-XRP amounts, the nested field names MUST be lower-case. If the `tfPartialPayment` flag is set, deliver _up to_ this amount instead.|
+|`DeliverMin`| |`string`|`Amount`|The minimum amount of destination currency this transaction should deliver. Only valid if this is a partial payment. For non-XRP amounts, the nested field names are lower-case.|
+|`Destination`|✔️|`string`|`AccountID`|The unique address of the account receiving the payment.|
+|`DestinationTag`| |`number`|`UInt32`|An arbitrary tag that identifies the reason for the payment to the destination, or a hosted recipient to pay.|
+|`InvoiceID`| |`string`|`Hash256`|An arbitrary 256-bit hash representing a specific reason or identifier for this payment.|
+|`Paths`| |`array`|`PathSet`|Array of payment paths to be used for this transaction. Must be omitted for XRP-to-XRP transactions.|
+|`SendMax` | |`string` or `object`|`Amount`|Highest amount of source currency this transaction is allowed to cost, including transfer fees, exchange rates, and [slippage](http://en.wikipedia.org/wiki/Slippage_%28finance%29). Does not include the XRP destroyed as a cost for submitting the transaction. For non-XRP amounts, the nested field names MUST be lower-case. Must be supplied for cross-currency/cross-issue payments. Must be omitted for XRP-to-XRP payments.|
+</details>
+
+We propose these additions:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`DomainID`| |`string`|`Hash256`|The domain that the offer must be a part of.|
+
+#### 5.1.1. `DomainID`
+
+The `DomainID` can only be included if the payment is a cross-currency payment (i.e. if the payment is going to interact with the DEX). It should only be included if the payment is permissioned.
+
+#### 5.1.2. `Flags`
+
+This spec proposes an additional flag, to indicate whether the offer should consider only the `DomainID`'s Permissioned DEX or also include the open DEX.
+
+### 5.2. Failure Conditions
+
+The existing set of failure conditions for `Payment` will continue to exist.
+
+There will also be the following in addition, if the `DomainID` is included:
+* The payment isn't a cross-currency payment.
+* The domain doesn't exist.
+* The `Account` is not a domain member.
```

Comment:
how do we check if an account is a domain member?

Reply:
does it work like:
1. look up the PermissionedDomain object
2. iterate through the AcceptedCredentials array
3. check if the `Subject` of the Credential object matches the account submitting the `Payment` tx?

Reply:
1. Look up the `PermissionedDomain` object
2. Iterate through the `AcceptedCredentials` array (which only contains issuer-credential type pairs)
3. Check if the (subject, issuer, credential type) credential exists (where the `Subject` is the account submitting the `Payment` tx)